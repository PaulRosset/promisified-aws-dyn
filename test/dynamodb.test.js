const AWS = require("aws-sdk");
const AWSUtils = require("./../lib/crudProcessing");

const table1 = {
  AttributeDefinitions: [
    {
      AttributeName: "Artist",
      AttributeType: "S"
    },
    {
      AttributeName: "SongTitle",
      AttributeType: "S"
    }
  ],
  KeySchema: [
    {
      AttributeName: "Artist",
      KeyType: "HASH"
    },
    {
      AttributeName: "SongTitle",
      KeyType: "RANGE"
    }
  ],
  ProvisionedThroughput: {
    ReadCapacityUnits: 5,
    WriteCapacityUnits: 5
  },
  TableName: "Music"
};
let dyn;

beforeAll(() => {
  dyn = new AWSUtils({
    endpoint: new AWS.Endpoint("http://localhost:8000"),
    accessKeyId: process.env.ACCESSKEYID,
    secretAccessKey: process.env.KEYSECRETAWS,
    region: "eu-west-2"
  });
});

describe("Testing tables", () => {
  test("Testing Creation table Music", async done => {
    try {
      const res = await dyn.createTable(table1);
      expect(res).toBeTruthy();
      expect(res.TableDescription.TableName).toEqual("Music");
      done();
    } catch (e) {
      if (e) throw e;
    }
  });
  test("Deleting table Music", async done => {
    try {
      const res = await dyn.deleteTable({ TableName: "Music" });
      expect(res).toBeTruthy();
      expect(res.TableDescription.TableName).toEqual("Music");
      done();
    } catch (e) {
      if (e) throw e;
    }
  });
});
