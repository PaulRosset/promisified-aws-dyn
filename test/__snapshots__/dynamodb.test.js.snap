// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add entry in Base Select item previously added 1`] = `
Object {
  "Item": Object {
    "Artist": Object {
      "S": "John Doe",
    },
    "SongTitle": Object {
      "S": "Hello World",
    },
  },
}
`;

exports[`Add entry in Base Update item previously added 1`] = `
Object {
  "Item": Object {
    "Artist": Object {
      "S": "John Doe",
    },
    "SongTitle": Object {
      "S": "Hello World",
    },
    "Year": Object {
      "S": "2089",
    },
  },
}
`;
